diff --git a/build.gradle b/build.gradle
index 991f5cb6..36c0b1bc 100644
--- a/build.gradle
+++ b/build.gradle
@@ -33,6 +33,8 @@ dependencyUpdates.resolutionStrategy {
 allprojects {
     tasks.withType(JavaForkOptions) {
         jvmArgs '-Djava.awt.headless=true'
+        // Add exports needed by Spark on newer JDKs (JDK 17/21).
+        jvmArgs '--add-exports=java.base/sun.nio.ch=ALL-UNNAMED'
     }
     repositories {
         mavenCentral()
@@ -719,14 +721,14 @@ task gitMergeAll {
 
 task run(type: JavaExec) {
     dependsOn checkRuntime, allBuildTasks, cleanTempDir
-    workingDir = '.'; jvmArgs = ['-server', '-XX:-OmitStackTraceInFastThrow']
+    workingDir = '.'; jvmArgs '-server', '-XX:-OmitStackTraceInFastThrow'
     systemProperties = ['moqui.conf':moquiConfDev, 'moqui.runtime':moquiRuntime]
     // NOTE: this is a hack, using -jar instead of a class name, and then the first argument is the name of the jar file
     mainClass = '-jar'; args = [warName]
 }
 task runProduction(type: JavaExec) {
     dependsOn checkRuntime, allBuildTasks, cleanTempDir
-    workingDir = '.'; jvmArgs = ['-server', '-Xms1024M']
+    workingDir = '.'; jvmArgs '-server', '-Xms1024M'
     systemProperties = ['moqui.conf':moquiConfProduction, 'moqui.runtime':moquiRuntime]
     mainClass = '-jar'; args = [warName]
 }
@@ -735,25 +737,25 @@ task load(type: JavaExec) {
     description = "Run Moqui to load data; to specify data types use something like: gradle load -Ptypes=seed,seed-initial,install"
     dependsOn checkRuntime, allBuildTasks
     systemProperties = ['moqui.conf':moquiConfDev, 'moqui.runtime':moquiRuntime]
-    workingDir = '.'; jvmArgs = ['-server']; mainClass = '-jar'
+    workingDir = '.'; jvmArgs '-server'; mainClass = '-jar'
     args = [warName, 'load', (project.properties.containsKey('types') ? "types=${types}" : "types=all")]
 }
 task loadSeed(type: JavaExec) {
     dependsOn checkRuntime, allBuildTasks
     systemProperties = ['moqui.conf':moquiConfProduction, 'moqui.runtime':moquiRuntime]
-    workingDir = '.'; jvmArgs = ['-server']; mainClass = '-jar'
+    workingDir = '.'; jvmArgs '-server'; mainClass = '-jar'
     args = [warName, 'load', (project.properties.containsKey('types') ? "types=${types}" : "types=seed")]
 }
 task loadSeedInitial(type: JavaExec) {
     dependsOn checkRuntime, allBuildTasks
     systemProperties = ['moqui.conf':moquiConfProduction, 'moqui.runtime':moquiRuntime]
-    workingDir = '.'; jvmArgs = ['-server']; mainClass = '-jar'
+    workingDir = '.'; jvmArgs '-server'; mainClass = '-jar'
     args = [warName, 'load', (project.properties.containsKey('types') ? "types=${types}" : "types=seed,seed-initial")]
 }
 task loadProduction(type: JavaExec) {
     dependsOn checkRuntime, allBuildTasks
     systemProperties = ['moqui.conf':moquiConfProduction, 'moqui.runtime':moquiRuntime]
-    workingDir = '.'; jvmArgs = ['-server']; mainClass = '-jar'
+    workingDir = '.'; jvmArgs '-server'; mainClass = '-jar'
     args = [warName, 'load', (project.properties.containsKey('types') ? "types=${types}" : "types=seed,seed-initial,install")]
 }
 
